symtabAPI/src/Object-pe.C

```diff
@@ -109,5 +215,57 @@ bool ObjectPE::isSharedLibrary() const
 
 Dyninst::Architecture ObjectPE::getArch() const
 {
-    return Arch_none;
+    if (!parsed_object_) return Arch_none;
+    switch (((peparse::parsed_pe *)parsed_object_)->peHeader.nt.FileHeader.Machine) {
+        case peparse::IMAGE_FILE_MACHINE_AMD64:
+            return Arch_x86_64;
+        case peparse::IMAGE_FILE_MACHINE_I386:
+            return Arch_x86;
+        case peparse::IMAGE_FILE_MACHINE_ARM:
+        case peparse::IMAGE_FILE_MACHINE_ARMNT:
+            return Arch_aarch32;
+        case peparse::IMAGE_FILE_MACHINE_ARM64:
+            return Arch_aarch64;
+        case peparse::IMAGE_FILE_MACHINE_UNKNOWN:
+        case peparse::IMAGE_FILE_MACHINE_ALPHA:
+        case peparse::IMAGE_FILE_MACHINE_ALPHA64:
+        case peparse::IMAGE_FILE_MACHINE_CEE:
+        case peparse::IMAGE_FILE_MACHINE_CEF:
+        case peparse::IMAGE_FILE_MACHINE_EBC:
+        case peparse::IMAGE_FILE_MACHINE_IA64:
+        case peparse::IMAGE_FILE_MACHINE_M32R:
+        case peparse::IMAGE_FILE_MACHINE_MIPS16:
+        case peparse::IMAGE_FILE_MACHINE_MIPSFPU:
+        case peparse::IMAGE_FILE_MACHINE_MIPSFPU16:
+        case peparse::IMAGE_FILE_MACHINE_POWERPC:
```

Comment:
The pe-parse comment says this is little-endian ppc, so let's mark it as `Arch_ppc64`.